clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py

```diff
@@ -64,18 +64,28 @@ def __init__(self, covid_reporter_key, internal_control_reporter_key):
 
     # TODO Later we might have to account for giving two trails over 38 as positive.
 
-    def _analyze_sample(self, sample):
+    # POS Control
+    # FAM=0 and VIC=0 => FAIL
+    # NEG Control
+    # FAM=0 and VIC=0 => NEG
+    # SAMPLE
+    # FAM=0 and VIC=0 => FAIL
+
+    def _analyze_sample(self, sample, is_neg_control=False):
         covid_ct = sample[self._covid_reporter_key]
         internal_control_ct = sample[self.internal_control_reporter_key]
         if covid_ct == 0 and internal_control_ct == 0:
-            return FAILED_BY_INTERNAL_CONTROL
+            if is_neg_control:
+                return COVID_RESPONSE_NEGATIVE
+            else:
+                return FAILED_BY_INTERNAL_CONTROL
         elif covid_ct == 0 and internal_control_ct <= self.INTERNAL_CONTROL_THRESHOLD:
             return COVID_RESPONSE_NEGATIVE
         elif covid_ct == 0 and internal_control_ct > self.INTERNAL_CONTROL_THRESHOLD:
             return FAILED_BY_INTERNAL_CONTROL
         elif covid_ct > self.COVID_CONTROL_THRESHOLD:
             return FAILED_BY_TOO_HIGH_COVID_VALUE
-        elif covid_ct <= self.COVID_CONTROL_THRESHOLD:
+        elif covid_ct <= self.COVID_CONTROL_THRESHOLD and covid_ct > 0:
```

Comment:
I'd suggest to have the lower bound on the left, perhaps (but not necessarily) using Python's syntactic sugar:

`elif 0 < covid_ct <= self.COVID_CONTROL_THRESHOLD`

Reply:
Will fix.